tests/test_routes.py

```diff
@@ -92,6 +96,54 @@ def test_index(self):
         resp = self.client.get("/")
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
 
+    def test_get_shopcart(self):
+        """It should Read a single Shopcart"""
+        # get the id of a shopcart
+        test_shopcart = self._create_an_empty_shopcart(1)[0]
+        resp = self.client.get(
+            f"{BASE_URL}/{test_shopcart.id}", content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+        self.assertEqual(data["id"], test_shopcart.id)
+
+    def test_get_shopcart_not_found(self):
+        """It should not Read a Shopcart that is not found"""
+        response = self.client.get(f"{BASE_URL}/0")
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+    
+   
+    def test_create_shopcarts(self):
+        """ It should create a shopcart """
+        shopcart = ShopcartFactory()
+        resp = self.client.post(
+            f"{BASE_URL}",json = shopcart.serialize(), content_type = "application/json"
+        )
+        
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+        
+        location = resp.headers.get("location")
+        #self.assertIsNotNone(location,None)
+ 
+        new_shopcart = resp.get_json()
+        # self.assertEqual(data["date"], shopcart.date,"date does not match")
+        self.assertEqual(new_shopcart["name"],shopcart.name, "name does not match")
+        
+
+    def test_create_shopcart_missing_info(self):
```

Comment:
this seems to be a test for create_shopart, and according to Professor's comment in another channel, I feel that we should put it in another PR 🤔

Reply:
ok